### PR TITLE
feat: microk8s bootstrap

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: <link>

### Overview

- Provide images w/ microk8s bootstrapped

### Rationale

- To install microk8s and pre-bootstrap w/ juju if installed.

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] The application version in `pyproject.toml` is incremented

### CODE REVIEW

File: src/github_runner_image_builder/templates/cloud-init.sh.j2 [hunk 1]
```diff
@@ -67,9 +67,8 @@ function configure_system_users() {
     /usr/bin/id -u ubuntu &>/dev/null || useradd --create-home ubuntu
     echo "PATH=\$PATH:/home/ubuntu/.local/bin" >> /home/ubuntu/.profile
     echo "PATH=\$PATH:/home/ubuntu/.local/bin" >> /home/ubuntu/.bashrc
-    /usr/sbin/groupadd -f microk8s
     /usr/sbin/groupadd -f docker
-    /usr/sbin/usermod --append --groups docker,microk8s,lxd,sudo ubuntu
```
Comment (jdkandersson) [inline src/github_runner_image_builder/templates/cloud-init.sh.j2:72]: Should these be here even if microk8s is not in the image?

Comment (yanksyoon) [reply src/github_runner_image_builder/templates/cloud-init.sh.j2:72]: You're right, nice catch :)

**Review by jdkandersson — APPROVED** [review]

File: tests/integration/test_openstack_builder.py [hunk 2]
```diff
@@ -99,6 +99,7 @@ def image_ids_fixture(
         image_config=config.ImageConfig(
             arch=image_config.arch,
             base=config.BaseImage.from_str(image_config.image),
+            microk8s="1.31-strict/stable",
```
Comment (alithethird) [inline tests/integration/test_openstack_builder.py:102]: Just out of curiosity, why use different MicroK8s version in unit and integration tests?

**Review by alithethird — APPROVED** [review]
1 nitpick but overall looks good to me. LGTM :rocket:

File: tests/integration/test_openstack_builder.py (hunk 2 above)
Comment (yanksyoon) [reply tests/integration/test_openstack_builder.py:102]: No particular reason, diversifying different versions :D

**Pull request merged**